src/overload_hessian.jl

```diff
@@ -1,85 +1,193 @@
 ## 1-to-1
-for fn in ops_1_to_1
-    @eval function Base.$fn(t::T) where {T<:GlobalHessianTracer}
-        if is_seconder_zero_global($fn)
-            if is_firstder_zero_global($fn)
-                return empty(T)
-            else
-                return t
-            end
+function hessian_tracer_1_to_1(
+    t::T, is_firstder_zero::Bool, is_seconder_zero::Bool
+) where {T<:HessianTracer}
+    if is_seconder_zero
+        if is_firstder_zero
+            return empty(T)
         else
-            return T(t.grad, t.hess ∪ (t.grad × t.grad))
+            return t
```

Comment:
I wonder if we shouldn't call the gradient tracer utilities in places like here

Reply:
because the non-zeroness of the first derivative can also play a role

Reply:
You're saying we should replace
```Julia
if is_seconder_zero
    if is_firstder_zero
        return empty(T)
    else
        return t
    end
else
    return T(gradient(t), hessian(t) ∪ (gradient(t) × gradient(t)))
end
```

by 
```Julia
if is_seconder_zero
    if is_firstder_zero
        return empty(T)
    else
        return t
    end
else
    if is_firstder_zero
        return T(empty(G), gradient(t) × gradient(t))
    else
        return T(gradient(t), hessian(t) ∪ (gradient(t) × gradient(t)))
    end
end
```
right?

Reply:
I'm saying Hessian tracing is a strict superset of gradient tracing, so everything we do for gradient tracing we need to re-do here too.
For code clarity I think it might help to split the construction of the `gradient` and `hessian` parts of the output tracer

Reply:
> For code clarity I think it might help to split the construction of the gradient and hessian parts of the output tracer

Skipping this for now. We can always revisit it later.